### PR TITLE
tests/util: Fix #4223: intermittent failing uchild tests

### LINKED ISSUE #4223: `test_uchild_when_run_no_wait_with_a_non_blocking_util` & `test_uchild_when_run_no_wait_with_a_blocking_command` are failing intermittently
```

---- common::util::tests::test_uchild_when_run_no_wait_with_a_non_blocking_util stdout ----
current_directory_resolved: 
run: /Users/runner/work/coreutils/coreutils/target/debug/coreutils echo hello world
---- common::util::tests::test_uchild_when_run_no_wait_with_a_non_blocking_util stderr ----
thread 'main' panicked at 'assertion failed: !child.is_alive()', tests/common/util.rs:2554:9
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

---- common::util::tests::test_uchild_when_run_no_wait_with_a_blocking_command stdout ----
current_directory_resolved: 
mkdir: /var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/.tmpxyJ9sh/a
touch: /var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/.tmpxyJ9sh/a/empty
run: /Users/runner/work/coreutils/coreutils/target/debug/coreutils rm -riv a
---- common::util::tests::test_uchild_when_run_no_wait_with_a_blocking_command stderr ----
thread 'main' panicked at 'Command was expected to succeed.
stdout = 
 stderr = ', tests/common/util.rs:177:9
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace


failures:
    common::util::tests::test_uchild_when_run_no_wait_with_a_blocking_command
    common::util::tests::test_uchild_when_run_no_wait_with_a_non_blocking_util
```

can be seen here:
https://github.com/uutils/coreutils/actions/runs/3642012657/jobs/6148626869

@Joining7943 could you please have a look?

### PR DESCRIPTION
**Pull request by Joining7943**
This pr increases the delay used in the `uchild` tests but also takes other measures to make these tests less prone to such intermittent failures.

Closes #4223

### CODE REVIEW

**Comment by Joining7943** [comment]
Could you review, please?

File: tests/common/util.rs [hunk 1]
```diff
@@ -2548,9 +2548,12 @@ mod tests {
     fn test_uchild_when_run_no_wait_with_a_non_blocking_util() {
         let ts = TestScenario::new("echo");
         let mut child = ts.ucmd().arg("hello world").run_no_wait();
-        child.delay(500);
 
-        // check `child.is_alive()` is working
+        // check `child.is_alive()` and `child.delay()` is working
+        while child.is_alive() {
```
Comment (tertsdiepraam) [inline tests/common/util.rs:2553]: Do we maybe want to limit this if the process is unresponsive? E.g. try 10 times and then panic?

Comment (Joining7943) [reply tests/common/util.rs:2553]: although I would have preferred to set a timeout ;)

**Review by tertsdiepraam — APPROVED** [review]
Alright, let's hope this fixes the issue!

**Pull request merged**